Squirtle ! Squirtle ! Squirtle !

RegisCoaxans opened the pull request:
## Description
Added Squad Leader Squirtle from the anime, to the game. It's a level evo using black glasses.

## Motivation and Context
Ash's Squirtle is very cool, we had to add it !

## How Has This Been Tested?
Loaded the game and obtained the Pokémon as intended.

## Types of changes
- New Pokémon

RegisCoaxans committed:
Squirtle ! Squirtle ! Squirtle !

RegisCoaxans committed:
Remove unused code

CypherX commented on src/scripts/items/HeldItem.ts:
```diff
@@ -251,4 +251,11 @@ ItemList.Everstone = new HeldItem('Everstone', 10000, GameConstants.Currency.mon
+    }
+);
+
+// Override Black Glasses' canUse function to factor in Squirtle evolution.
+const glassesRestrict = (ItemList.Black_Glasses as AttackBonusHeldItem).canUse;
+(ItemList.Black_Glasses as AttackBonusHeldItem).canUse = (pokemon: PartyPokemon): boolean => {
+    return pokemon.name != 'Squad Leader Squirtle' ? pokemon.name == 'Squirtle' || glassesRestrict(pokemon) : false;
+};
```
I'm not exactly thrilled with doing it this way but the only alternatives I can think of are having `TypeRestrictedAttackBonusHeldItem` accept an optional function parameter for additional criteria that `canUse` calls, or extend `TypeRestrictedAttackBonusHeldItem` with a new class to do something similar

Qwertypop04 replied:
I think it also feels unintuitive, although maybe that's not a bad thing, since it's an optional Pokémon.

aheadfrogg commented:
Why was level 25 selected?

RedzepR commented:
Instead of placing this Pokemon in the FS it should spawn at R11 to mimic the fact you get Squirtle in Pokemon Yellow in Viridian/Lets Go.
<img width="1280" height="720" alt="image" src="https://github.com/user-attachments/assets/6a83f970-b1bd-442a-874f-e774d5c3af95" />
 
Then I'd place an NPC (Officer Jenny) in Viridian either once you move to Johto or obtain your first Black Glasses that says something along the lines "A bunch of wild squirtles terrorize the area, it would need a strong squirtle leader to unite them... i heard their love sunglasses." because having an optional mon with no hint ingame is whack.

And then I'd also like to have the squirtle *squad* as a pokemon, not just the leader. 
Preferably in their later anime outfit.
<img width="584" height="328" alt="image" src="https://github.com/user-attachments/assets/5ace5b08-a808-4bdd-ba09-e704463b21aa" />

Askellown commented:
> Instead of placing this Pokemon in the FS it should spawn at R11 to mimic the fact you get Squirtle in Pokemon Yellow in Viridian/Lets Go. 

I think in the anime in happens between the Charmander episode and the Bill's lighthouse episode meaning it's on 24/25

RegisCoaxans commented:
> Why was level 25 selected?

No particular reason. I wanted a greater level than 16, but lower than 36. 25 looked like a good middleground.

RegisCoaxans committed:
Level 10, route EV after capture, Jenny hint

RegisCoaxans committed:
Converted exception into new held item class

RegisCoaxans committed:
Merge branch 'develop' into SquadLeader

RegisCoaxans commented:
Added to route 11 for EV sake.
Officer Jenny is added to Vermilion and tells the player about the Squirtle Squad.
Changed the level evolution from 25 to 10 so it can be triggered before evolving into Wartortle.

RegisCoaxans committed:
eslint

RegisCoaxans committed:
Merge branch 'SquadLeader' of https://github.com/RegisCoaxans/pokeclicker into SquadLeader
